File: trains-gdx-core/src/main/java/com/github/caay2000/trains/world/Route.java
package com.github.caay2000.trains.world;

import com.github.caay2000.trains.common.Position;

public class Route {

    private WorldObject start;
    private WorldObject end;

    public Route(WorldObject start, WorldObject end) {
        this.start = start;
        this.end = end;
    }

    public WorldObject getStart() {
        return start;
    }

    public WorldObject getEnd() {
        return end;
    }
}

File: trains-gdx-core/src/main/kotlin/com/github/caay2000/trains/world/Route.kt
package com.github.caay2000.trains.world

import com.github.caay2000.trains.world.location.City

class Route(var start: City, var end: City) {

    fun endRoute() {
        this.start = this.end.also { this.end = this.start }
    }

    fun nextStop() = this.end
}